exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,2329 @@
+	if err != nil {
+		t.Error(err)
+	}
+	if _, err = d.WSRetriveDeposits(currencyBTC, 25, 0); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetWithdrawals(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := d.GetWithdrawals(context.Background(), currencyBTC, 25, 0)
+	if err != nil {
+		t.Error(err)
+	}
+	if _, err := d.WSRetriveWithdrawals(currencyBTC, 25, 0); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitTransferToSubAccount(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or canManipulateRealOrders isnt set correctly")
+	}
+	_, err := d.SubmitTransferToSubAccount(context.Background(), currencyBTC, 0.01, 13434)
+	if err != nil && !strings.Contains(err.Error(), "transfer_not_allowed") {
+		t.Error(err)
+	}
+	if _, err = d.WSSubmitTransferToSubAccount(currencyBTC, 0.01, 13434); err != nil && !strings.Contains(err.Error(), "transfer_not_allowed") {
+		t.Error(err)
+	}
+}
+
+func TestSubmitTransferToUser(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or canManipulateRealOrders isnt set correctly")
+	}
+	_, err := d.SubmitTransferToUser(context.Background(), currencyBTC, "", "13434", 0.001)
+	if err != nil && !strings.Contains(err.Error(), "transfer_not_allowed") {
+		t.Error(err)
+	}
+	if _, err = d.WSSubmitTransferToUser(currencyBTC, "", "0x4aa0753d798d668056920094d65321a8e8913e26", 0.001); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitWithdraw(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or canManipulateRealOrders isnt set correctly")
+	}
+	_, err := d.SubmitWithdraw(context.Background(), currencyBTC, core.BitcoinDonationAddress, "", 0.001)
+	if err != nil {
+		t.Error(err)
+	}
+	if _, err = d.WSSubmitWithdraw(currencyBTC, core.BitcoinDonationAddress, "", 0.001); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetAnnouncements(t *testing.T) {
+	t.Parallel()
+	_, err := d.GetAnnouncements(context.Background(), time.Now(), 5)
+	if err != nil {
+		t.Error(err)
+	}
+	if _, err = d.WSRetriveAnnouncements(time.Now(), 5); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetPublicPortfolioMargins(t *testing.T) {
+	t.Parallel()
+	info, err := d.GetInstrumentData(context.Background(), "BTC-PERPETUAL")
+	if err != nil {
+		t.Skip(err)
+	}
+	if _, err = d.GetPublicPortfolioMargins(context.Background(), currencyBTC, map[string]float64{
+		"BTC-PERPETUAL": info.ContractSize * 2,
+	}); err != nil {
+		t.Error(err)
+	}
+}
+func TestWSRetrivePublicPortfolioMargins(t *testing.T) {
+	t.Parallel()
+	info, err := d.GetInstrumentData(context.Background(), "BTC-PERPETUAL")
+	if err != nil {
+		t.Skip(err)
+	}
+	if _, err = d.WSRetrivePublicPortfolioMargins(currencyBTC, map[string]float64{
+		"BTC-PERPETUAL": info.ContractSize * 2,
+	}); err != nil {
+		t.Error(err)
+	}
```

Comment:
Can merge with test above like the others

Reply:
yea! As for other methods, we can not call these two functions at a close time range. It raises an error.